Source/src/core/GameObject.cpp

```diff
@@ -265,10 +299,15 @@ void Hachiko::GameObject::UpdateBoundingBoxes()
     }
 }
 
-void Hachiko::GameObject::RemoveComponent(Component* component)
+bool Hachiko::GameObject::AttemptRemoveComponent(Component* component)
 {
     //TODO: Should I delete the component?
-    components.erase(std::remove(components.begin(), components.end(), component));
+    if (component->CanBeRemoved())
+    {
+        components.erase(std::remove(components.begin(), components.end(), component));
```

Comment:
We have to check if this is not creating a memory leak, since the erase will only remove the pointer from the vector, but won't release the memory